### ISSUE TITLE
EditorField instances not ordered according to DOM tree after adding more fields

### ISSUE THREAD

**Issue by kleinerpirat** [ankitects/anki#1980]
https://github.com/ankitects/anki/blob/50e36bc312b64c59fcd53c283596daeeaba9521e/ts/sveltelib/lifecycle-hooks.ts#L41-L42

Mounting new fields (e.g. when switching from Basic to Basic + reverse card) shifts the initial fields to the end of `EditorField.instances`:

![image](https://user-images.githubusercontent.com/62722460/180394761-28c04277-746f-419c-bc46-bb3a7e158ffb.png)
![image](https://user-images.githubusercontent.com/62722460/180394900-018026dc-a91e-4cf5-af87-23a3833bd027.png)
![image](https://user-images.githubusercontent.com/62722460/180394948-8d2b588c-0ea5-4dff-b5db-1d45d587b728.png)

Developers will assume the instances are ordered according to their appearance in the DOM tree, so we should handle the case where instances is already filled with something like:

```
instances = oldInstances.concat(newInstances);
```

It seems to me however, that the `setup` function of `lifecycleHooks` is agnostic to the difference between mounting fields when loading the first note and appending new fields when switching to another note. Any ideas?

**Comment by hgiesel** [comment]
You're right, it doesn't make sense to call unshift here. But the comment hints that if two additional fields are mounted, the hook will be first called for the latter one first, so that is also something to keep in mind. I don't know if we can accomodate that at the moment?

> The setup function of lifecycleHooks seems to be oblivious to the bigger picture (in this case, the difference between mounting fields when loading the first note and appending new fields when switching to another note). Any ideas?

Yes, the lifecycle hooks only work in "Svelte" logic. So if you switch notetypes, and both have 2 fields, the hook won't even be fired. This can be useful for setup code. For example, if you want to additional UI elements, this would be better done here. As for notetype-switching logic, this would use another kind of hook, which I've implemented here:

https://github.com/ankitects/anki/pull/1691/files#diff-f837972ae4746bfa1bb14c5e4b77fefa43d7f672709bc8744dc1061c13cca71eR95-R100

Unfortunately, the PR is still on ice :)

**Comment by dae** [comment]
Sorry it's been such a delay :-( Hopefully not too much longer.
